### PR TITLE
R4023 Documentation - RPaaS PUT async rule 

### PR DESCRIPTION
**Pull request by sanjaiganesh**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: documentation/openapi-authoring-automated-guidelines.md
Comment (akning-ms) [inline documentation/openapi-authoring-automated-guidelines.md:147]: Is this rule similar as R2005 ?

Comment (sanjaiganesh) [reply documentation/openapi-authoring-automated-guidelines.md:147]: No. This is RPaaS specific and validates only PUT. I agree there is a bit of overlap.
R2005 is actually not correct. Is this rule active ?
Even 202 is allowed long running status code for PUT, DELETE & POST. 

_**Description**: The allowed response status codes for a long DELETE operation are "200", "204". The allowed response status codes for a POST operation are "200", "201" & "204". The allowed response status codes for a PUT operation are  "200" & "201"._


_**R4014 (RPaaS specific rule)**_ 
- Only 201 is supported
-x-ms-long-running-operation added (lot of RPs miss this today)
-  x-ms-long-running-operation-options is specified

Comment (sanjaiganesh) [reply documentation/openapi-authoring-automated-guidelines.md:152]: also updated to use R4023

Comment (sanjaiganesh) [reply documentation/openapi-authoring-automated-guidelines.md:152]: you may want to update R2005 to include 202 as well. It is not checking x-ms-* stuff. it is just validating status code.

Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md:152]: Also similar as R2010 which is used for validating POST operation, but make sense

**Review by jianyexi — APPROVED** [review]

**Review by josefree — APPROVED** [review]

**Comment by jianyexi** [comment]
Please do not merge it until the code of the rule been merged

**Comment by akning-ms** [comment]
> Please do not merge it until the code of the rule been merged

Pls let me know if it is ready for merge. @sanjaiganesh @jianyexi

**Pull request merged**
